### ISSUE TITLE
Can not find different between 2 SHA in the current branch

### ISSUE THREAD

**Issue by kateluu-comdev** [scolladon/sfdx-git-delta#156]
Issue verification check :

- In github action ( workflow), set up the flow to connect to my target org ( successfully) , then find the delta between last commit with a specific commit ( identify by SHA)
here is  my command to find out delta between 2 commits in the same (current) branch, with b9ce097 is SHA of commit in the current branch
sfdx sgd:source:delta --from b9ce097 --output .

## What is the problem
I expected 2 files ( 1 apex, 1 cmp) change which commit at SHA but the result show as blank 
 ```
 sfdx sgd:source:delta --from b9ce097 --output .
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
{
  "error": null,
  "output": ".",
  "success": true,
  "warnings": []
}
```

and when I checked the package.xml file, here is the result
 ```
 echo "-- package.xml generated --" | cat package/package.xml
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
<?xml version="1.0" encoding="UTF-8"?>
<Package xmlns="http://soap.sforce.com/2006/04/metadata">
    <version>52.0</version>
</Package>
```
Can team please preview what is missing in my command or what command I should use to achieve my result if the one I am using is not appropriate. 

Cheers,

**Comment by mehdisfdc** [comment]
Hi @kateluu-comdev! Thank you for logging your issue!

To get a better understanding of your issue, could you run a a git diff in the same context, and let us know the result?
`git diff --name-only b9ce097`

Thank you!

**Comment by scolladon** [comment]
Hi @kateluu-comdev !

What is the configuration of the github action ?
Does it contains a checkout with a [fetch-depth](https://github.com/actions/checkout#checkout-v2) value ?

Duplicate of #142

**Comment by scolladon** [comment]
Duplicate of #142

**Comment by kateluu-comdev** [comment]
@mehdisfdc @scolladon  Hi Mehdisfdc and Scolladon , 
Thank for your response and here is the outcome when I run the **git diff command** on my context
![gitdiff](https://user-images.githubusercontent.com/78332307/123561394-6d8a5200-d7eb-11eb-8a92-42085d185d41.jpg)
Content: 
```
$ git diff --name-only b9ce097
.github/workflows/DeltaDeploy.yml
force-app/main/default/lwc/riskyCustomer/riskyCustomer.html
```
And there is the **workflow file** which I use to test my my delta command
```
# This is a basic workflow to help you get started with Actions

name: CI with Delta

# Controls when the workflow will run
on:
  # Triggers the workflow on push or pull request events but only for the master branch
  push:
    branches: 
      - sfdx
  
  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:
    

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
 
  # This workflow contains a single job called "build-deploy"
  build-deploy-delta:
        # The type of runner that the job will run on
        runs-on: ubuntu-latest

        # Steps represent a sequence of tasks that will be executed as part of the job
        steps:
            # Install Salesforce CLI
            - name: 'Install Salesforce CLI'
              run: |
                  wget https://developer.salesforce.com/media/salesforce-cli/sfdx/channels/stable/sfdx-linux-x64.tar.xz
                  mkdir ~/sfdx
                  tar xJf sfdx-linux-x64.tar.xz -C ~/sfdx --strip-components 1
                  echo "$HOME/sfdx/bin" >> $GITHUB_PATH
                  ~/sfdx/bin/sfdx version
         

            # Install Git Delta
            - name: 'Install Git Delta'
              run: echo y | sfdx plugins:install sfdx-git-delta
            
            # Checkout the source code
            - name: 'Checkout source code'
              uses: actions/checkout@v2

            # Compare and pickup delta change only
            - name: 'Compare and pickup delta change only'
              shell: bash
              run: sfdx sgd:source:delta --from b9ce097 --output .
              
            # Logout package.xml file
            - name: 'Logout package.xml file'
              shell: bash
              run: echo "-- package.xml generated --" | cat package/package.xml

            # Goodbuy  Message
            - name: 'Logout status'
              if: always()
              run: echo "Complete with OK status"
```
Please let me know if you need any further information to assist me. Thanks

Cheers,

**Comment by scolladon** [comment]
Hi @kateluu-comdev !

Thanks for the information.

I think the issue here is related to the checkout fetch-depth.
According to the [doc](https://github.com/actions/checkout), the configuration should specify it should be set to 0 to get all the repo.

Could you give it a try and let us know ?

**Comment by kateluu-comdev** [comment]
Hi @scolladon ,

Thank a lot for the clue, I did try again with fetch-depth and it works perfectly. Can you please close the ticket. Cheers. Thank again for the support.

**Comment by kateluu-comdev** [comment]
Hi @scolladon ,

Thank a lot for the clue, I did try again with fetch-depth and it works perfectly. Thank again for the support. Cheers,
Close the ticket for now
